streamflow-api.yaml

```diff
@@ -0,0 +1,768 @@
+          example:
+            code: '401'
+            message: Message giving details of authorization problem
+
+    404NotFound:
+      description: The specified resource was not found.
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/Error'
+          example:
+            code: '404'
+            message: Resource not found
+
+    409Conflict:
+      description: Attempt to put resource into impossible, inconsistent, or unavailable state.
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/Error'
+          example:
+            code: '409'
+            message: Message giving details of conflict
+
+  schemas:
+
+    ### Generic schemas
+
+    Error:
+      description: Generic error response body
+      type: object
+      properties:
+        code:
+          type: string
+        message:
+          type: string
+      required:
+        - code
+        - message
+      example:
+        code: '404'
+        message: Message giving details of problem
+
+    URI:
+      description: URI property
+      type: object
+      properties:
+        uri:
+          type: string
+
+    HypermediaLink:
+      description: Annotated hypermedia link to a related resource
+      type: object
+      properties:
+        rel:
+          type: string
+        uri:
+          type: string
+      required:
+        - rel
+        - uri
+
+    HypermediaLinkList:
+      description: List of annotated hypermedia links to related resources
+      type: array
+      items:
+        $ref: '#/components/schemas/HypermediaLink'
+
+    Timeseries:
+      description: A time series, which is to say a set of values at specified times.
+        Data points are (time[i], value[i]).
+      type: object
+      properties:
+        times:
+          description: Time coordinates, specified as ISO8601 date-time strings.
+          type: array
+          items:
+            type: string
+          example: ["2000-01-15", "2000-02-15"]
+        values:
+          type: array
+          items:
+            type: number
+          example: [5000, 6000]
+      required:
+        - times
+        - values
+
+    ### Hydromodel outputs
+
+    # Individual hydromodel outputs
+
+    HydromodelOutputId:
+      description: Unique id of hydromodel output that drives a streamflow computation
+      type: integer
+      example: 4
+
+    HydromodelOutputState:
+      description: State for a hydromodel output. Contains complete state.
+      allOf:
+        - type: object
+          properties:
+            hydromodel_output_id:
+              $ref: '#/components/schemas/HydromodelOutputId'
+            model:
+              type: string
+              example: CanESM2
+            experiment:
+              type: string
+              example: rcp85
+            etc:
+              type: string
+              example: ... more tbd ...
```

Comment:
Spatial coverage info is needed, I think, so that it's possible to tell whether a result for a given point can be generated from this hydromodel output. 

Something like minimum and maximum latitude and longitude, as well as latitude and longitude cell size would be enough to provide context to the cell_x and cell_y values in the result endpoint.

Reply:
Yuck, it looks like hydromodel output files have irregular spatial extents:
![spatialcoverage](https://user-images.githubusercontent.com/4512605/44162610-ab476200-a075-11e8-9830-d8bc12a088c8.jpg)

So in order to provide enough information to determine whether a result file for  given point can be made from this hydromodel output, we may need to return a geoJSON polygon or something as the spatial output., not just a bounding box. That's a bit of a headache.

Reply:
Hm, maybe if we need to provide actual spatial coverage, it could be a separate resource, like maybe, like `/streamflow/hydromodel_output/4/bounds`? We probably don't want to calculate it until a client is specifically looking for it, though caching could be a big help, it's not like the coverage of a netCDF file is going to change.

Reply:
I agree completely. Caching will make a very big difference here, but it still may be that it's better to provide the bounds at a separate endpoint. At this stage, I prefer that. 

(BTW, the bounds would likely be a kind of information not supplied in the list resource that would be present in the hypothetical item resources. But partitioning it as a separate endpoint eliminates that question.)

Reply:
Will we want to add filtering (to the list resource) by location? E.g., list all hydromodels that contain this location? Or some spatial extent?

Reply:
Hm. Do you think it's better to do that sort of filtering in the back end or the front end?